NTumbleBit/ClassicTumbler/Client/PaymentStateMachine.cs

```diff
@@ -200,23 +198,30 @@ public CycleProgressInfo Update()
 					CyclePhase.ClientCashoutPhase
 				};
 
-                if (cycle.IsComplete(height))
-                    return null;
-
-                //If we are not in any phase we are in the SaftyPeriod
-                if (!phases.Any(p => cycle.IsInPhase(p, height)))
-                {
-                    phase = lastCyclePhase;
-                    period = cycle.GetPeriods().GetPeriod(phase);
-                    period.End += cycle.SafetyPeriodDuration;
-                    isSaftyPeriod = true;
-                }
-                else
-                {
-                    phase = phases.First(p => cycle.IsInPhase(p, height));
-                    period = cycle.GetPeriods().GetPeriod(phase);
-                    isSaftyPeriod = false;
-                }
+				if (cycle.IsComplete(height))
+					return null;
+
+				//If we are not in any phase we are in the SaftyPeriod.
+				if (!phases.Any(p => cycle.IsInPhase(p, height)))
+				{
+					//Find last CyclePhase 
+					for (int i = height - 1; i >= cycle.Start; i--)
+					{
+						if (phases.Any(p => cycle.IsInPhase(p, i)))
```

Comment:
This loop does appear to be searching, but is it changing anything? There are no variable/state changes, so it seems to have no effect